libclamav/scanners.c

```diff
@@ -1703,6 +1709,11 @@ static cl_error_t cli_ole2_tempdir_scan_vba_new(const char *dir, cli_ctx *ctx, s
         close(tempfd);
         tempfd = -1;
     }
+    if (tempfile && !ctx->engine->keeptmp) {
```

Comment:
Only removing the tempfile should be wrapped with the `keeptmp` check. We still need to free the variable and set to NULL either way. 
E.g. like is done in `xlm_extract.c`:
```c
    if (tempfile && !ctx->engine->keeptmp) {
        remove(tempfile);
    }
    FREE(tempfile);
```